components/lcd/esp_lcd_ili9881c/include/esp_lcd_ili9881c.h

```diff
@@ -10,25 +10,118 @@
 
 #pragma once
 
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#if SOC_MIPI_DSI_SUPPORTED
 #include "esp_lcd_panel_vendor.h"
+#include "esp_lcd_mipi_dsi.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @brief Create LCD control panel for ILI9881C
+ * @brief LCD panel initialization commands.
+ *
+ */
+typedef struct {
+    int cmd;                /*<! The specific LCD command */
+    const void *data;       /*<! Buffer that holds the command specific data */
+    size_t data_bytes;      /*<! Size of `data` in memory, in bytes */
+    unsigned int delay_ms;  /*<! Delay in milliseconds after this command */
+} ili9881c_lcd_init_cmd_t;
+
+/**
+ * @brief LCD panel vendor configuration.
+ *
+ * @note  This structure needs to be passed to the `vendor_config` field in `esp_lcd_panel_dev_config_t`.
+ *
+ */
+typedef struct {
+    const ili9881c_lcd_init_cmd_t *init_cmds;       /*!< Pointer to initialization commands array. Set to NULL if using default commands.
+                                                     *   The array should be declared as `static const` and positioned outside the function.
+                                                     *   Please refer to `vendor_specific_init_default` in source file.
+                                                     */
+    uint16_t init_cmds_size;                        /*<! Number of commands in above array */
+    struct {
+        esp_lcd_dsi_bus_handle_t dsi_bus;               /*!< MIPI-DSI bus configuration */
+        const esp_lcd_dpi_panel_config_t *dpi_config;   /*!< MIPI-DPI panel configuration */
+        uint8_t  lane_num;                              /*!< Number of MIPI-DSI lanes */
+    } mipi_config;
```

Comment:
should we wrap it inside a union? so we can extend to support other interface in the future.

Reply:
There's no need to create a union for the ILI9881, as it only supports the MIPI-DSI interface.